pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -364,6 +364,14 @@ func (c *Controller) syncService(key string) error {
 		return err
 	}
 
+	// Drop EndpointSlices that have been marked for deletion to prevent the controller from getting stuck.
+	for i, endpointSlice := range endpointSlices {
+		if endpointSlice.DeletionTimestamp != nil {
+			endpointSlices[i] = endpointSlices[len(endpointSlices)-1]
+			endpointSlices = endpointSlices[:len(endpointSlices)-1]
+		}
+	}
```

Comment:
this panics in some scenarios https://go.dev/play/p/Rd8SpdjIAli
better use the recommended "filter in place" for slices https://github.com/golang/go/wiki/SliceTricks#filter-in-place